src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -165,14 +165,12 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
 
   override def reverse = {
     val r = new ArrayDeque[A](initialSize = size)
-    var i = length
-    while(0 < i) {
-      i -= 1
-      r.appendAssumingCapacity(this(i))
-    }
+    this.foreach(r.prependAssumingCapacity)
```

Comment:
It's eliding the closure completely that's likely to be the hardest work for the JVM, not that it is a while loop.  Usually there are a couple more levels of indirection which need to be worked through.  It has the best chance of doing so in a microbenchmark, but even then it often can't manage.